core/ledger/kvledger/snapshot_mgmt_test.go

```diff
@@ -0,0 +1,128 @@
+	kvl := l.(*kvLedger)
+
+	bcInfo, _ := l.GetBlockchainInfo()
+	require.Equal(t, &common.BlockchainInfo{
+		Height: 1, CurrentBlockHash: gbHash, PreviousBlockHash: nil,
+	}, bcInfo)
+
+	// add requests and verify smallestRequestHeight
+	require.NoError(t, kvl.snapshotRequestBookkeeper.add(100))
+	require.Equal(t, uint64(100), kvl.snapshotRequestBookkeeper.smallestRequestHeight)
+
+	require.NoError(t, kvl.snapshotRequestBookkeeper.add(15))
+	require.Equal(t, uint64(15), kvl.snapshotRequestBookkeeper.smallestRequestHeight)
+
+	require.NoError(t, kvl.snapshotRequestBookkeeper.add(50))
+	require.Equal(t, uint64(15), kvl.snapshotRequestBookkeeper.smallestRequestHeight)
+
+	requestHeights, err := kvl.snapshotRequestBookkeeper.list()
+	require.NoError(t, err)
+	require.ElementsMatch(t, requestHeights, []uint64{15, 50, 100})
+
+	for _, height := range []uint64{15, 50, 100} {
+		exist, err := kvl.snapshotRequestBookkeeper.exist(height)
+		require.NoError(t, err)
+		require.True(t, exist)
+	}
+
+	exist, err := kvl.snapshotRequestBookkeeper.exist(10)
+	require.NoError(t, err)
+	require.False(t, exist)
+
+	provider.Close()
+
+	// reopen the provider and ledger and verify snapshotRequestBookkeeper is initialized correctly
+	provider2 := testutilNewProvider(conf, t, &mock.DeployedChaincodeInfoProvider{})
+	defer provider2.Close()
+	l2, err := provider2.Open(ledgerID)
+	require.NoError(t, err)
+	kvl2 := l2.(*kvLedger)
+
+	requestHeights, err = kvl2.snapshotRequestBookkeeper.list()
+	require.NoError(t, err)
+	require.ElementsMatch(t, requestHeights, []uint64{15, 50, 100})
+
+	require.Equal(t, uint64(15), kvl2.snapshotRequestBookkeeper.smallestRequestHeight)
+
+	// delete requests and verify smallest request height
+	require.NoError(t, kvl2.snapshotRequestBookkeeper.delete(100))
+	require.Equal(t, uint64(15), kvl2.snapshotRequestBookkeeper.smallestRequestHeight)
+
+	require.NoError(t, kvl2.snapshotRequestBookkeeper.delete(15))
+	require.Equal(t, uint64(50), kvl2.snapshotRequestBookkeeper.smallestRequestHeight)
+
+	require.NoError(t, kvl2.snapshotRequestBookkeeper.delete(50))
+	require.Equal(t, defaultSmallestHeight, kvl2.snapshotRequestBookkeeper.smallestRequestHeight)
+
+	requestHeights, err = kvl2.snapshotRequestBookkeeper.list()
+	require.NoError(t, err)
+	require.ElementsMatch(t, requestHeights, []uint64{})
+}
+
+func TestSnapshotRequestBookKeeper_ErrorPaths(t *testing.T) {
+	conf, cleanup := testConfig(t)
+	defer cleanup()
+	provider := testutilNewProvider(conf, t, &mock.DeployedChaincodeInfoProvider{})
+	defer provider.Close()
+
+	ledgerID := "testsnapshotrequestbookkeeper-errorpaths"
+	_, gb := testutil.NewBlockGenerator(t, ledgerID, false)
+	gbHash := protoutil.BlockHeaderHash(gb.Header)
+	l, err := provider.Create(gb)
+	require.NoError(t, err)
+	defer l.Close()
+	kvl := l.(*kvLedger)
+
+	bcInfo, _ := l.GetBlockchainInfo()
+	require.Equal(t, &common.BlockchainInfo{
+		Height: 1, CurrentBlockHash: gbHash, PreviousBlockHash: nil,
+	}, bcInfo)
```

Comment:
Though this code is not related and you could directly create an instance. However, since you have it here, it will be good to add a case for error propagation during ledger opening.